Sming/Libraries/MultipartParser/src/PartCheckerStream.h

```diff
@@ -0,0 +1,79 @@
+/****
+ * Sming Framework Project - Open Source framework for high efficiency native ESP8266 development.
+ * Created 2015 by Skurydin Alexey
+ * http://github.com/SmingHub/Sming
+ * All files of the Sming Core are provided under the LGPL v3 license.
+ *
+ * PartCheckerStream.h
+ *
+ * @author: 2021 - Slavey Karadzhov <slav@attachix.com>
+ *
+ ****/
+
+#pragma once
+
+#include <Data/Stream/StreamWrapper.h>
+#include <Network/Http/HttpHeaders.h>
+#include <Delegate.h>
+
+class PartCheckerStream : public StreamWrapper
+{
+public:
+	struct FilePart {
+		String name;
+		String fileName;
+		String mime;
+		int length; // -1 if not available, >0 otherwise
+	};
+
+	using CheckerCallback =
+		Delegate<bool(const HttpHeaders& headers, ReadWriteStream* source, const FilePart& filePart)>;
+
+	/**
+	 * @param callback
+	 * @param stream - The actual stream doing the work. The stream is owned and will be deleted here
+	 */
+	PartCheckerStream(CheckerCallback callback, ReadWriteStream* source) : StreamWrapper(source), callback(callback)
+	{
+	}
+
+	bool checkHeaders(const HttpHeaders& headers, const FilePart& part)
+	{
+		save = callback(headers, source, part);
+		return save;
+	}
+
+	bool isSuccess()
+	{
+		return save;
+	}
+
```

Comment:
Do we need another method here? Streams have an `isValid` virtual method which seems appropriate.